### PR TITLE
enhance: Support Array DataType for bulk_insert

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: https://github.com/milvus-io/milvus/issues/28272
master pr: #28341 
Support array DataType for bulk_insert with json, binlog files.

### CODE REVIEW

**Comment by xiaocai2333** [comment]
rerun ut

File: internal/util/importutil/binlog_file.go [hunk 1]
```diff
@@ -478,6 +478,49 @@ func (p *BinlogFile) ReadJSON() ([][]byte, error) {
...
+		}
+
+		if p.DataType() != schemapb.DataType_Array {
+			log.Warn("Binlog file: binlog data type is not Array")
+			return nil, errors.New("binlog data type is not Array")
+		}
+
+		data, err := event.PayloadReaderInterface.GetArrayFromPayload()
+		if err != nil {
+			log.Warn("Binlog file: failed to read JSON data", zap.Error(err))
```
Comment (czs007) [inline internal/util/importutil/binlog_file.go:514]: failed to read Array data

**Pull request merged**